src/main/java/net/atos/zac/app/planitems/PlanItemsRESTService.java

```diff
@@ -115,22 +113,20 @@ public void doHumanTask(final RESTHumanTaskData humanTaskData) {
     public void doUserEventListener(final RESTUserEventListenerData userEventListenerData) {
         switch (userEventListenerData.actie) {
             case INTAKE_AFRONDEN -> {
+                final PlanItemInstance planItemInstance = caseService.readOpenPlanItem(userEventListenerData.planItemInstanceId);
+                caseVariablesService.setOntvankelijk(planItemInstance.getCaseInstanceId(), userEventListenerData.zaakOntvankelijk);
                 if (!userEventListenerData.zaakOntvankelijk) {
                     final Zaak zaak = zrcClientService.readZaak(userEventListenerData.zaakUuid);
                     final ZaakafhandelParameters zaakafhandelParameters = zaakafhandelParameterService.readZaakafhandelParameters(zaak);
-                    if (zaakafhandelParameters.getNietOntvankelijkResultaattype() == null) {
-                        throw new ResulttaattypeNotFoundException("geen resultaattype voor het niet ontvankelijk verklaren");
-                    }
                     zgwApiService.createResultaatForZaak(zaak, zaakafhandelParameters.getNietOntvankelijkResultaattype(),
                                                          userEventListenerData.resultaatToelichting);
                 }
-                caseService.startUserEventListener(userEventListenerData.planItemInstanceId);
             }
             case ZAAK_AFHANDELEN -> {
                 zgwApiService.createResultaatForZaak(userEventListenerData.zaakUuid, userEventListenerData.resultaattypeUuid,
                                                      userEventListenerData.resultaatToelichting);
-                caseService.startUserEventListener(userEventListenerData.planItemInstanceId);
             }
         }
+        caseService.startUserEventListener(userEventListenerData.planItemInstanceId);
```

Comment:
Is dit wel juist? Willen we UserEventListeners starten die bij ons niet bekend zijn? Want deze staat nu buiten de switch.

Ze zouden toch eens een finally variant in de switches moeten inbouwen ;)